File: pracaInż/Models/Entities/Employee.cs
using pracaInż.Models.DTO.Employees;
using pracaInż.Models.Entities;
using pracaInż.Models.Entities.CompanyStructure;
using System;

public class Employee
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Surname { get; set; }
    public string EmailAddress { get; set; }
    public string WorkPhone { get; set; }
    public string JobTitle { get; set; }

    public int DepartmentId { get; set; }   
    public Department Department { get; set; }

    public List<Software> SoftwareAccessList { get; set; }

    public Employee()
    {

    }

    public Employee(AddEmployeeBasiInfoDTO employeeDTO, Department department)
    {
        Name = employeeDTO.Name;
        Surname = employeeDTO.Surname;
        EmailAddress = employeeDTO.Email;
        WorkPhone = employeeDTO.WorkPhoneNumber;
        JobTitle = employeeDTO.JobTitle;
        Department = department;
        DepartmentId = department.Id;
    }

    public Employee(UpdateEmployeeDTO employeeDTO, Department department)
    {
        Id = employeeDTO.Id;
        Name = employeeDTO.Name;
        Surname = employeeDTO.Surname;
        EmailAddress = employeeDTO.Email;
        WorkPhone = employeeDTO.WorkPhoneNumber;
        JobTitle = employeeDTO.JobTitle;
        Department = department;
        DepartmentId = department.Id;
    }


}

File: pracaInż/Frotnend/src/app/Models/Employee.ts
import { Computer } from "./Computer";
import { Department } from "./Department";
import { Factory } from "./Factory";

export interface Employee {
    id: number;
    name: string;
    surname: string;
    email: string;
    workPhoneNumber: string;
    jobTitle: string;
    departmentName: string;
    departmentShortName: string;
    factoryLocation: string;
    computer: Computer
}

export interface EmployeePC {
    id: number;
    name: string;
    lastName: string;
}